demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,511 @@
+        unsigned rightClass;
+        std::chrono::time_point<std::chrono::steady_clock> startTime;
+    };
+
+    InferRequest &inferRequest;
+    std::vector<InferRequestImage> images;
+};
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------------Parsing and validation of input args----------------------------------
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+
+    slog::info << "Parsing input parameters" << slog::endl;
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+    if (FLAGS_labels.empty()) {
+        throw std::logic_error("Parameter -labels is not set");
+    }
+    if (FLAGS_nt < 1) {
+        throw std::runtime_error("Parameter -nt must be >= 1");
+    }
+
+    return true;
+}
+
+void resizeImage(cv::Mat& image, int modelInputResolution) {
+    double scale = static_cast<double>(modelInputResolution) / std::min(image.cols, image.rows);
+    cv::resize(image, image, cv::Size(), scale, scale);
+
+    cv::Rect imgROI;
+    if (image.cols >= image.rows) {
+        int fromWidth = image.cols/2 - modelInputResolution/2;
+        imgROI = cv::Rect(fromWidth,
+                          0,
+                          std::min(modelInputResolution, image.cols - fromWidth),
+                          modelInputResolution);
+    } else {
+        int fromHeight = image.rows/2 - modelInputResolution/2;
+        imgROI = cv::Rect(0,
+                          fromHeight,
+                          modelInputResolution,
+                          std::min(modelInputResolution, image.rows - fromHeight));
+    }
+    image(imgROI).copyTo(image);
+}
+
+std::vector<std::vector<unsigned>> topResults(Blob& input, unsigned int n = 5) {
+    std::vector<std::vector<unsigned>> output;
+    using currentBlobType = PrecisionTrait<Precision::FP32>::value_type;
+    TBlob<currentBlobType>& tblob = dynamic_cast<TBlob<currentBlobType>&>(input);
+    size_t batchSize =  tblob.getTensorDesc().getDims()[0];
+    std::vector<unsigned> indices(tblob.size() / batchSize);
+    n = static_cast<unsigned>(std::min<size_t>((size_t) n, tblob.size()));
+    output.resize(batchSize);
+    for (size_t i = 0; i < batchSize; i++) {
+        size_t offset = i * (tblob.size() / batchSize);
+        currentBlobType *batchData = tblob.data();
+        batchData += offset;
+        std::iota(std::begin(indices), std::end(indices), 0);
+        std::partial_sort(std::begin(indices), std::begin(indices) + n, std::end(indices),
+                          [&batchData](unsigned l, unsigned r) {
+                              return batchData[l] > batchData[r];
+                          });
+        for (unsigned j = 0; j < n; j++) {
+            output.at(i).push_back(indices.at(j));
```

Comment:
After a partial sort you can do like
```
indices.resize(n);
output[i] = std::move(indices);
```
It requires moving `std::vector<unsigned> indices(tblob.size() / batchSize);` inside of  the loop.